src/Calculator/Controls/CalculationResult.cpp

```diff
@@ -377,7 +377,16 @@ void CalculationResult::OnTapped(TappedRoutedEventArgs ^ e)
 
 void CalculationResult::OnRightTapped(RightTappedRoutedEventArgs ^ e)
 {
-    this->Focus(::FocusState::Programmatic);
+    auto requestedElement = e->OriginalSource;
+
+    if (requestedElement->Equals(safe_cast<Object ^>(m_textBlock)))
```

Comment:
use dynamic_cast instead since m_textBlock could technically be nullptr if a style is applied to this control that doesn't; include it.

Reply:
Makes sense! I will do the change.